### ISSUE TITLE
Pop-up window not centered with latest version 0.0.20 : FIXED

### ISSUE THREAD

**Issue by aquarime** [u-Sir/drag-to-preview#5]
Hi,

Facing centered pop-up issue with latest version 0.0.20
Option `Try to open at mouse position` is **_unchecked_**
Pop-up width and height are respected but is not centered.
I had faced this issue with version 0.0.17, issue had been fixed with 0.0.19, reappears oddly with 0.0.20

Release notes for 0.0.20 on the extension's AMO page state:
- ensure that popup window centered based on current screen when tryOpenAtMousePosition is false.  
Seems that, at least in my case, the enforcement to center the pop-up is the culprit ...

**Comment by u-Sir** [comment]
is it not centered based on window or screen ?
in 0.0.19, popup window is centered based on window.

**Comment by aquarime** [comment]
> is it not centered based on window?

I'm not sure I quite understand your question.
Firefox window here is maximized. Displayed normally and resized to whatever dimensions, the pop-up remains non centered.

**Comment by u-Sir** [comment]
When the option "Try to open at mouse position" is unchecked, if the Firefox window isn't maximized, the popup window will appear like this picture:

<img width="1710" alt="图片" src="https://github.com/user-attachments/assets/8b612a8b-8935-48ac-854e-75b998f3c1a1">

In version 0.0.19, the popup window will always appear inside the Firefox window.

If you want to ensure the popup window doesn't display outside the original window, I will add options to set this in the next version.

**Comment by u-Sir** [comment]
it's weird.
"Try to open at mouse position" is unchecked and maximized Firefox window  on my devices is centered. 😂
<img width="1710" alt="图片" src="https://github.com/user-attachments/assets/6f3ecddc-086b-4937-ae99-bf03161b3ffb">

**Comment by aquarime** [comment]
Weird indeed. Here, whatever the window's size, the pop-up is never centered.
I've tried resizing the window to a whatever size and dragged a link, here's my screenshot:

![1](https://github.com/user-attachments/assets/f0e36058-8a13-4f28-a655-ec4e0173f8fd)

I'm no coder, all I can do is report my experience.

**Comment by u-Sir** [comment]
> Weird indeed. Here, whatever the window's size, the pop-up is never centered. I've tried resizing the window to a whatever size and dragged a link, here's my screenshot:
> 
> ![1](https://private-user-images.githubusercontent.com/175529702/348526795-f0e36058-8a13-4f28-a655-ec4e0173f8fd.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA5NDg5MTMsIm5iZiI6MTcyMDk0ODYxMywicGF0aCI6Ii8xNzU1Mjk3MDIvMzQ4NTI2Nzk1LWYwZTM2MDU4LThhMTMtNGYyOC1hNjU1LWVjNGUwMTczZjhmZC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxNFQwOTE2NTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NWZhODQyYTcwNTY2ZTg1OWIwNDk4YmNkNzcxMzVhMmI0NzIyMDViZDY2ZTU2ZDYyMGE2Nzg2M2U1YjM4MTc1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.5OB7eD-jblYvjFf3r8_PGpu1d69_fZg34D9CJft7W0A)
> 
> I'm no coder, all I can do is report my experience.


![图片](https://github.com/user-attachments/assets/cbb79bdf-e7cb-45a4-bb43-13ce2b0c612c)


in your pic, popup window is centered based on current screen, same as my screenshot.   
In next version, I will add options to set this.

**Comment by aquarime** [comment]
> in your pic, popup window is centered based on current screen, same as my screenshot.

No it is not! What is centered in my pic is the Firefox window, not the pop-up. I've clicked on the Firefox window to have it appear in front. The pop-up is in back, top-left ...

**Comment by u-Sir** [comment]
please use this web to check screen info: https://www.whatismyscreenresolution.org/

**Comment by u-Sir** [comment]
and would you mind load this debug version on your side?



    1. type "about:addons" on address bar and enter ,and disabled current extension

    2. download this and extract it.:

    3. type "about:debugging#/runtime/this-firefox" on address bar and enter
[drag-to-preview-0.0.20-firefox-test.zip](https://github.com/user-attachments/files/16216502/drag-to-preview-0.0.20-firefox-test.zip)


    4. click on "load temporary add-on" and select the manifest.json where you extracted.
    5. after extension loaded, click on "inspect" near the extension.
    6. click "console" on inspect window
    7. drag on normal page
    8. screenshot like this
<img width="762" alt="图片" src="https://github.com/user-attachments/assets/daf82673-d307-4be8-a282-f97460ae79a0">

**Comment by aquarime** [comment]
https://www.whatismyscreenresolution.org/  reports erroneous values, as does i.e. https://www.deviceinfo.me/ > Screen
This has always surprised me given I do NOT use Firefox's anti-fingerprinting pref :` pref privacy.resistFingerprinting` is **false** (default).
I linger to know how Firefox manages to provide these erroneous values. Maybe a default anti-fingerprinting policy, whether `pref privacy.resistFingerprinting` be set to _false_ (default) or _true_ ...

_What Is My Screen Resolution_ reports:

Screen Resolution width : 1664px : WRONG
Screen Resolution height : 936px : WRONG
Device Resolution: 1920 x 1080 : CORRECT
Device inner Resolution: 1920 x 978 : CORRECT
Browser Viewport: 1664 x 848 : WRONG

**Comment by aquarime** [comment]
Debug version drag-to-preview-0.0.20-firefox-test - Firefox 115.13.0 ESR console results:

```
Received message in background script: 
Object { checkContextMenuItem: true }
background.js:73:13
Current window: 
Object { id: 1, focused: true, top: -4, left: -4, width: 1928, height: 1088, incognito: false, type: "normal", state: "maximized", alwaysOnTop: false, … }
background.js:86:17
Config saved: lastClientX = undefined background.js:33:21
Config saved: lastClientY = undefined background.js:33:21
Config saved: lastScreenTop = undefined background.js:33:21
Config saved: lastScreenLeft = undefined background.js:33:21
Config saved: lastScreenWidth = undefined background.js:33:21
Config saved: lastScreenHeight = undefined background.js:33:21
Received message in background script: 
Object { checkContextMenuItem: true }
background.js:73:13
Current window: 
Object { id: 1, focused: true, top: -4, left: -4, width: 1928, height: 1088, incognito: false, type: "normal", state: "maximized", alwaysOnTop: false, … }
background.js:86:17
Config saved: lastClientX = undefined background.js:33:21
Config saved: lastClientY = undefined background.js:33:21
Config saved: lastScreenTop = undefined background.js:33:21
Config saved: lastScreenLeft = undefined background.js:33:21
Config saved: lastScreenWidth = undefined background.js:33:21
Config saved: lastScreenHeight = undefined background.js:33:21
Received message in background script: 
Object { linkUrl: "https://www.theregister.com/2024/07/12/korea_star_wars/?td=rt-3a", lastClientX: 946, lastClientY: 171, width: 1664, height: 936, top: 0, left: 0 }
background.js:73:13
Current window: 
Object { id: 1, focused: true, top: -4, left: -4, width: 1928, height: 1088, incognito: false, type: "normal", state: "maximized", alwaysOnTop: false, … }
background.js:86:17
Config saved: lastClientX = 946 background.js:33:21
Config saved: lastClientY = 171 background.js:33:21
Config saved: lastScreenTop = 0 background.js:33:21
Config saved: lastScreenLeft = 0 background.js:33:21
Config saved: lastScreenWidth = 1664 background.js:33:21
Config saved: lastScreenHeight = 936 background.js:33:21
Config saved: originWindowId = 1 background.js:33:21
tryOpenAtMousePosition: false background.js:188:13
lastScreenLeft: 0 background.js:189:13
lastScreenTop: 0 background.js:190:13
lastScreenWidth: 1664 background.js:191:13
lastScreenHeight: 936 background.js:192:13
dx - dy : 32-3.5 background.js:193:13
Received message in background script: 
Object { checkContextMenuItem: true }
background.js:73:13
Current window: 
Object { id: 1322, focused: true, top: 4, left: 32, width: 1600, height: 929, incognito: false, type: "popup", state: "normal", alwaysOnTop: false, … }
background.js:86:17
Config saved: lastClientX = undefined background.js:33:21
Config saved: lastClientY = undefined background.js:33:21
Config saved: lastScreenTop = undefined background.js:33:21
Config saved: lastScreenLeft = undefined background.js:33:21
Config saved: lastScreenWidth = undefined background.js:33:21
Config saved: lastScreenHeight = undefined background.js:33:21
Received message in background script: 
Object { checkContextMenuItem: true }
background.js:73:13
Current window: 
Object { id: 1, focused: true, top: -4, left: -4, width: 1928, height: 1088, incognito: false, type: "normal", state: "maximized", alwaysOnTop: false, … }
background.js:86:17
Config saved: lastClientX = undefined background.js:33:21
Config saved: lastClientY = undefined background.js:33:21
Config saved: lastScreenTop = undefined background.js:33:21
Config saved: lastScreenLeft = undefined background.js:33:21
Config saved: lastScreenWidth = undefined background.js:33:21
Config saved: lastScreenHeight = undefined background.js:33:21
Received message in background script: 
Object { checkContextMenuItem: true }
background.js:73:13
Current window: 
Object { id: 1, focused: true, top: -4, left: -4, width: 1928, height: 1088, incognito: false, type: "normal", state: "maximized", alwaysOnTop: false, … }
background.js:86:17
Config saved: lastClientX = undefined background.js:33:21
Config saved: lastClientY = undefined background.js:33:21
Config saved: lastScreenTop = undefined background.js:33:21
Config saved: lastScreenLeft = undefined background.js:33:21
Config saved: lastScreenWidth = undefined background.js:33:21
Config saved: lastScreenHeight = undefined background.js:33:21
Received message in background script: 
Object { checkContextMenuItem: true }
background.js:73:13
Current window: 
Object { id: 1322, focused: true, top: 4, left: 32, width: 1600, height: 929, incognito: false, type: "popup", state: "normal", alwaysOnTop: false, … }
background.js:86:17
Config saved: lastClientX = undefined background.js:33:21
Config saved: lastClientY = undefined background.js:33:21
Config saved: lastScreenTop = undefined background.js:33:21
Config saved: lastScreenLeft = undefined background.js:33:21
Config saved: lastScreenWidth = undefined background.js:33:21
Config saved: lastScreenHeight = undefined background.js:33:21
```

**Comment by u-Sir** [comment]
what's your browser/system zoom level?

**Comment by aquarime** [comment]
> what's your browser/system zoom level?

115% with a few exceptions

**Comment by u-Sir** [comment]
> > what's your browser/system zoom level?
> 
> 115% with a few exceptions

Thanks. I can reproduced it on my device now. Will fixed it in next version. ☺️

**Comment by aquarime** [comment]
I've just tested with FF Zoom to 100% and pop-up is indeed centered now.
Nice find :)

**Comment by u-Sir** [comment]
Please check if it fixed in 0.0.21 on AMO.

**Comment by aquarime** [comment]
Just updated to 0.0.21, dragged links with different zoom levels: FIXED :+1: 
I experience no issue for the time being with this latest 0.0.21, but I'll find one, lol (this IS a joke!).
Nice work, thanks for your commitment. There were issues, you worked and solved them. Great, for a great extension which is so handy :)
